### ISSUE TITLE
java.util.NoSuchElementException: null

### ISSUE THREAD

**Issue by ornicar** [ReactiveMongo/ReactiveMongo#731]
This is logged a bunch of times every 10 seconds in prod. `172.16.0.2:27017` is a secondary in a replicaset that has 4 secondaries.

```
2018-01-27 14:29:42,285 WARN  r.core.actors.MongoDBSystem Fails to update history: ConnectAll$IsMaster(294, {{NodeSet None Node[172.16.0.2:27017: Secondary (10/10 available connections), latency=1], auth=Set() | Node[172.16.0.3:27017: Secondary (10/10 available connections), latency=1], auth=Set() | Node[172.16.0.5:27017: Primary (10/10 available connections), latency=0], auth=Set() }})
java.util.NoSuchElementException: null
        at java.util.ArrayDeque.removeFirst(ArrayDeque.java:280) ~[na:1.8.0_144]
        at java.util.ArrayDeque.remove(ArrayDeque.java:447) ~[na:1.8.0_144]
        at shaded.google.common.collect.EvictingQueue.add(EvictingQueue.java:107) ~[org.reactivemongo.reactivemongo-shaded-0.12.2.jar:0.12.2]
        at shaded.google.common.collect.EvictingQueue.offer(EvictingQueue.java:91) ~[org.reactivemongo.reactivemongo-shaded-0.12.2.jar:0.12.2]
        at reactivemongo.core.actors.MongoDBSystem$class.go$1(actors.scala:244) [org.reactivemongo.reactivemongo_2.11-0.12.2.jar:0.12.2]
        at reactivemongo.core.actors.MongoDBSystem$class.reactivemongo$core$actors$MongoDBSystem$$updateHistory(actors.scala:252) [org.reactivemongo.reactivemongo_2.11-0.12.2.jar:0.12.2]
        at reactivemongo.core.actors.MongoDBSystem$$anonfun$7$$anonfun$apply$8.apply(actors.scala:216) [org.reactivemongo.reactivemongo_2.11-0.12.2.jar:0.12.2]
        at reactivemongo.core.actors.MongoDBSystem$$anonfun$7$$anonfun$apply$8.apply(actors.scala:215) [org.reactivemongo.reactivemongo_2.11-0.12.2.jar:0.12.2]
        at reactivemongo.core.actors.MongoDBSystem$class.updateNodeSet(actors.scala:1058) [org.reactivemongo.reactivemongo_2.11-0.12.2.jar:0.12.2]
        at reactivemongo.core.actors.StandardDBSystem.updateNodeSet(actors.scala:1407) [org.reactivemongo.reactivemongo_2.11-0.12.2.jar:0.12.2]
        at reactivemongo.core.actors.MongoDBSystem$$anonfun$reactivemongo$core$actors$MongoDBSystem$$onIsMaster$1.apply$mcV$sp(actors.scala:1024) [org.reactivemongo.reactivemongo_2.11-0.12.2.jar:0.12.2]
        at akka.actor.Scheduler$$anon$4.run(Scheduler.scala:126) [com.typesafe.akka.akka-actor_2.11-2.4.20.jar:na]
        at akka.dispatch.TaskInvocation.run(AbstractDispatcher.scala:39) [com.typesafe.akka.akka-actor_2.11-2.4.20.jar:na]
        at akka.dispatch.ForkJoinExecutorConfigurator$AkkaForkJoinTask.exec(AbstractDispatcher.scala:415) [com.typesafe.akka.akka-actor_2.11-2.4.20.jar:na]
        at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260) [org.scala-lang.scala-library-2.11.12.jar:na]
        at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339) [org.scala-lang.scala-library-2.11.12.jar:na]
        at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979) [org.scala-lang.scala-library-2.11.12.jar:na]
        at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107) [org.scala-lang.scala-library-2.11.12.jar:na]
```

`rs.status()` says the secondary is doing good:

```
    {
      "_id": 12,
      "name": "172.16.0.3:27017",
      "health": 1,
      "state": 2,
      "stateStr": "SECONDARY",
      "uptime": 4474487,
      "optime": {
        "ts": Timestamp(1517063707, 383),
        "t": NumberLong("87")
      },
      "optimeDurable": {
        "ts": Timestamp(1517063707, 383),
        "t": NumberLong("87")
      },
      "optimeDate": ISODate("2018-01-27T14:35:07Z"),
      "optimeDurableDate": ISODate("2018-01-27T14:35:07Z"),
      "lastHeartbeat": ISODate("2018-01-27T14:35:07.328Z"),
      "lastHeartbeatRecv": ISODate("2018-01-27T14:35:05.985Z"),
      "pingMs": NumberLong("1"),
      "syncingTo": "172.16.0.5:27017",
      "configVersion": 228995
    },
    {
      "_id": 14,
      "name": "172.16.0.2:27017",
      "health": 1,
      "state": 2,
      "stateStr": "SECONDARY",
      "uptime": 14507550,
      "optime": {
        "ts": Timestamp(1517063707, 125),
        "t": NumberLong("87")
      },
      "optimeDurable": {
        "ts": Timestamp(1517063707, 125),
        "t": NumberLong("87")
      },
      "optimeDate": ISODate("2018-01-27T14:35:07Z"),
      "optimeDurableDate": ISODate("2018-01-27T14:35:07Z"),
      "lastHeartbeat": ISODate("2018-01-27T14:35:07.150Z"),
      "lastHeartbeatRecv": ISODate("2018-01-27T14:35:06.640Z"),
      "pingMs": NumberLong("0"),
      "syncingTo": "172.16.0.5:27017",
      "configVersion": 228995
    }
```

**Comment by cchantep** [comment]
Hi, this have been fixed by #723 and will be part of the next release (for now that a benign, annoying, warning that can be ignored).

**Comment by ornicar** [comment]
Awesome, thank you!
